packages/nitro-protocol/test/contracts/ninja-nitro/SingleChannelAdjudicator/claim.test.ts

```diff
@@ -0,0 +1,337 @@
+    ${'16. underspecified guarantee, overfunded      '} | ${{g: 12}} | ${['B']}              | ${{A: 5, B: 5}}       | ${[]}   | ${{A: 5, B: 0}}       | ${{g: 7}} | ${{B: 5}}       | ${undefined}
+  `(
+    '$name',
+    async ({
+      name,
+      heldBefore,
+      guaranteeDestinations,
+      tOutcomeBefore,
+      indices,
+      tOutcomeAfter,
+      heldAfter,
+      payouts,
+      reason,
+    }: {
+      name;
+      heldBefore: AssetOutcomeShortHand;
+      guaranteeDestinations;
+      tOutcomeBefore: AssetOutcomeShortHand;
+      indices: number[];
+      tOutcomeAfter: AssetOutcomeShortHand;
+      heldAfter: AssetOutcomeShortHand;
+      payouts: AssetOutcomeShortHand;
+      reason;
+    }) => {
+      const target = new TestChannel(getRandomNonce(name));
+      const guarantor = new TestChannel(getRandomNonce(name + 'g'));
+      addresses.t = target.id;
+      addresses.g = guarantor.id;
+      // Transform input data (unpack addresses and BigNumber amounts)
+      [heldBefore, tOutcomeBefore, tOutcomeAfter, heldAfter, payouts] = [
+        heldBefore,
+        tOutcomeBefore,
+        tOutcomeAfter,
+        heldAfter,
+        payouts,
+      ].map(object => replaceAddressesAndBigNumberify(object, addresses) as AssetOutcomeShortHand);
+      guaranteeDestinations = guaranteeDestinations.map(x => addresses[x]);
+      // Fund the guarantor channel
+      let gasUsed;
+      gasUsed = await guarantor.depositETH(heldBefore[guarantor.id]);
+      await writeGasConsumption(
+        'SingleChannelAdjudicator.claim.gas.md',
+        name + ': send ETH to Guarantor',
+        gasUsed
+      );
+      // DEPLOY GUARANTOR CHANNEL
+      gasUsed = await guarantor.deploy();
+      await writeGasConsumption(
+        'SingleChannelAdjudicator.claim.gas.md',
+        name + ': deploy Guarantor',
+        gasUsed
+      );
+      // Compute an appropriate guarantee for the guarantor (using only ETH)
+      const guarantee = {
+        destinations: guaranteeDestinations,
+        targetChannelId: target.id,
+      };
+      const guarantorOutcome: Outcome = [{assetHolderAddress: constants.AddressZero, guarantee}];
+      if (guaranteeDestinations.length > 0) {
+        // CONCLUDE GUARANTOR CHANNEL
+        gasUsed = await guarantor.conclude(guarantorOutcome);
+        await writeGasConsumption(
+          'SingleChannelAdjudicator.claim.gas.md',
+          name + ': conclude Guarantor',
+          gasUsed
+        );
+      } else guarantor.outcome = guarantorOutcome; // Set this so that the claim tx should revert in the way we expect
+      // Compute an appropriate allocation outcome for the target (using only ETH)
+      const allocation = [];
+      Object.keys(tOutcomeBefore).forEach(key =>
+        allocation.push({destination: key, amount: tOutcomeBefore[key]})
+      );
```

Comment:
Consider a semantically meaningful variable name here: eg:

```suggestion
      Object.keys(tOutcomeBefore).forEach(eoa =>
        allocation.push({destination: eoa, amount: tOutcomeBefore[eoa]})
      );
```
It took me some time to discern what `key` referred to in some cases on other test files. Especially important here because the shape of this object changes in the file.

Reply:
I went with `destination` in the end, since it is clear that is how `key` is being used in the side effect of the callback. 

As a side observation, `replacedAddresseAndBigNumberify`, as well as the `addresses` lookup object (prolific throughout the tests in this package) are poorly named. We should be using `destination` (32 bytes, a Nitro concept) instead of `address` (20 bytes, an Ethereum concept). Having said that, a future change we would likely want to make in ninja-nitro would be to largely deprecate `destinations`.

Reply:
eda43d0